src/main/java/org/folio/dao/impl/ConfigurationsDaoImpl.java

```diff
@@ -0,0 +1,269 @@
+package org.folio.dao.impl;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import org.apache.commons.lang3.builder.ReflectionDiffBuilder;
+import org.apache.commons.lang3.builder.DiffResult;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.folio.config.ConfigurationsClient;
+import org.folio.config.model.SamlConfiguration;
+import org.folio.dao.ConfigurationsDao;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.interfaces.Results;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.util.model.OkapiHeaders;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * @author barbaraloehle
+ */
+public class ConfigurationsDaoImpl implements ConfigurationsDao {
+  private static final Logger LOGGER = LogManager.getLogger(ConfigurationsDaoImpl.class);
+  private static final String ERROR_MESSAGE_STRING = "errorMessage : %s";
+
+  public static final String CONFIGURATION_TABLE = "configuration";
+
+  private Future<SamlConfiguration> getConfigurationMigration(Vertx vertx, OkapiHeaders okapiHeaders, boolean withDelete) {
+
+    Objects.requireNonNull(okapiHeaders);
+    ConfigurationsDao.verifyOkapiHeaders(okapiHeaders);
+    return PostgresClient.getInstance(vertx, okapiHeaders.getTenant())
+      .get(CONFIGURATION_TABLE, SamlConfiguration.class, new Criterion(), true)
+      .compose(results -> localFutureGetConfigurationMigration(results, vertx, okapiHeaders, withDelete));
+  }
+
+  private Future<SamlConfiguration> localFutureGetConfigurationMigration(Results<SamlConfiguration> results, Vertx vertx,
+    OkapiHeaders okapiHeaders, boolean withDelete) {
+
+    int localLength = results.getResults().size();
+    if (localLength == 1) {
+      return Future.succeededFuture(results.getResults().get(0));
+    }
+    if(localLength > 1) {
+      String errorMessage = String.format("Migration: Number of records are not unique. Instead the number is : %s",
+        Integer.toString(localLength));
+      LOGGER.error(ERROR_MESSAGE_STRING, errorMessage);
+      return Future.failedFuture(new IllegalArgumentException(errorMessage));
+    }
+    return ConfigurationsClient.getConfigurationWithIds(vertx, okapiHeaders)
+      .compose(result -> storeSamlConfiguration(vertx, okapiHeaders, result))
+      .compose(result -> {
+        if (withDelete) {
+          return localDeleteConfigurationEntries(vertx, okapiHeaders, result)
+            .compose(resultAfterDeletion -> Future.succeededFuture(result))
+            .onFailure(cause -> Future.failedFuture(cause.getMessage()));
```

Comment:
The line
.onFailure(cause -> Future.failedFuture(cause.getMessage()));
can be removed because the the failure is handled in the the method localDeleteConfigurationEntries
method.